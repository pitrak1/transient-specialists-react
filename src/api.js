import axios from 'axios'
import utils from './utils'

const attachInterceptors = () => {
  axios.interceptors.response.use(
    function(response) {
      return response.data.statusCode === 500
        ? Promise.reject(response)
        : response
    },
    function(error) {
      return Promise.reject(error)
    },
  )
}

const translateError = error => {
  if (error.includes('duplicate key value violates unique constraint')) {
    if (error.includes('idx_type_name')) {
      return 'Name must be unique (case insensitive)'
    }

    if (error.includes('idx_oem_name')) {
      return 'Name must be unique (case insensitive)'
    }

    if (error.includes('idx_model_name')) {
      return 'Name must be unique (case insensitive)'
    }

    if (error.includes('idx_equipment_serial_number')) {
      return 'Serial Number must be unique (case insensitive)'
    }
  }

  if (error.includes('violates foreign key constraint')) {
    if (error.includes('equipments_model_id_fkey')) {
      return 'Model cannot be deleted because it has related equipment'
    }

    if (error.includes('equipments_type_id_fkey')) {
      return 'Type cannot be deleted because it has related equipment'
    }

    if (error.includes('models_oem_id_fkey')) {
      return 'OEM cannot be deleted because it has related models'
    }
  }

  return error
}

const getIndex = (resource, options, success, failure) => {
  attachInterceptors()
  const { ascending, page, perPage, searchValue, sortBy } = options
  return axios
    .get(
      `${
        process.env.LAMBDA_ENDPOINT
      }${resource}?ascending=${ascending}&page=${page}&perPage=${perPage}&searchValue=${searchValue}&sortBy=${utils.pascalToSnake(
        sortBy,
      )}`,
    )
    .then(result => {
      success(utils.convertObject(result.data.body))
    })
    .catch(error => {
      failure(error.data.body)
    })
}

const getShow = (resource, id, options, success, failure) => {
  attachInterceptors()
  const { ascending, page, perPage, sortBy } = options
  return axios
    .get(
      `${
        process.env.LAMBDA_ENDPOINT
      }${resource}?id=${id}&ascending=${ascending}&page=${page}&perPage=${perPage}&sortBy=${utils.pascalToSnake(
        sortBy,
      )}`,
    )
    .then(result => {
      success(utils.convertObject(result.data.body))
    })
    .catch(error => {
      failure(error.data.body)
    })
}

const getNew = (resource, success, failure) => {
  attachInterceptors()
  return axios
    .get(`${process.env.LAMBDA_ENDPOINT}${resource}?new=true`)
    .then(result => {
      success(utils.convertObject(result.data.body))
    })
    .catch(error => {
      failure(error.data.body)
    })
}

export const postCreate = (resource, data, success, failure) => {
  attachInterceptors()
  return axios
    .post(`${process.env.LAMBDA_ENDPOINT}${resource}`, data)
    .then(response => {
      success(response.data.body)
    })
    .catch(error => {
      failure(translateError(error.data.body))
    })
}

export const patchUpdate = (resource, data, success, failure) => {
  attachInterceptors()
  return axios
    .patch(`${process.env.LAMBDA_ENDPOINT}${resource}`, data)
    .then(response => {
      success(response.data.body)
    })
    .catch(error => {
      failure(translateError(error.data.body))
    })
}

export const deleteDestroy = (resource, id, success, failure) => {
  attachInterceptors()
  return axios
    .delete(`${process.env.LAMBDA_ENDPOINT}${resource}?id=${id}`)
    .then(response => {
      success(response.data.body)
    })
    .catch(error => {
      failure(translateError(error.data.body))
    })
}

export default {
  getIndex,
  getShow,
  getNew,
  postCreate,
  patchUpdate,
  deleteDestroy,
}
