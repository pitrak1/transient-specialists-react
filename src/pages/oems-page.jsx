import React from 'react'
import {
  Button,
  CircularProgress,
  TextField,
  Typography,
} from '@material-ui/core'
import FullTable from '../components/full-table.jsx'
import { withRouter } from 'react-router'
import api from '../api.js'

export class OemsPage extends React.Component {
  state = {
    alert: null,
    ascending: true,
    data: {},
    error: null,
    loading: true,
    page: 0,
    perPage: 10,
    searchValue: '',
    sortBy: 'name',
  }

  componentDidMount() {
    this.getData()
  }

  getData = () => {
    this.setState({ loading: true })
    const { ascending, page, perPage, searchValue, sortBy } = this.state
    api.getIndex(
      'oems',
      {
        ascending,
        page,
        perPage,
        searchValue,
        sortBy,
      },
      result => {
        this.setState({ loading: false, data: result })
      },
      error => {
        this.setState({ loading: false, error })
      },
    )
  }

  handleShowEquipmentClick = id => {
    this.props.history.push(`/equipment/search/${id}`)
  }

  handleShowModelsClick = id => {
    this.props.history.push(`/models/search/${id}`)
  }

  handleDeleteClick = id => {
    if (confirm('Are you sure you want to delete this OEM?')) {
      this.setState({ loading: true })
      api.deleteDestroy(
        'oems',
        id,
        _response => this.getData,
        error => {
          this.setState({ loading: false, alert: error })
        },
      )
    }
  }

  handleAddClick = _event => {
    this.props.history.push(`/oems/create`)
  }

  handleSearchChange = (_e, value) => {
    this.setState({ searchValue: value })
  }

  handleSearchClick = () => {
    this.getData()
  }

  handlePageChange = (_event, newPage) => {
    this.setState({ page: newPage }, this.getData)
  }

  handlePerPageChange = event => {
    this.setState({ perPage: event.target.value }, this.getData)
  }

  handleSort = (sortBy, ascending) => {
    this.setState({ sortBy, ascending }, this.getData)
  }

  render() {
    if (this.state.loading) {
      return <CircularProgress />
    }

    if (this.state.error) {
      return <div>{this.state.error}</div>
    }

    const headers = [
      { type: 'value', id: 'name', label: 'Name' },
      { type: 'button', id: 'showEquipment' },
      { type: 'button', id: 'showModels' },
      { type: 'button', id: 'delete' },
    ]

    const data = this.state.data.oems.map(oem => ({
      id: oem.id,
      cells: [
        { id: 'name', type: 'value', value: oem.name },
        {
          id: 'showEquipment',
          type: 'button',
          value: 'Show Equipment',
          callback: this.handleShowEquipmentClick,
        },
        {
          id: 'showModels',
          type: 'button',
          value: 'Show Models',
          callback: this.handleShowModelsClick,
        },
        {
          id: 'delete',
          type: 'button',
          value: 'Delete',
          callback: this.handleDeleteClick,
        },
      ],
    }))

    return (
      <div>
        {this.state.alert && <div>{this.state.alert}</div>}
        <Typography variant='h5'>OEMs</Typography>
        <TextField
          id='search'
          label='Search'
          value={this.state.searchValue}
          onChange={this.handleSearchChange}
          variant='outlined'
        />
        <Button onClick={this.handleSearchClick}>Search</Button>
        <Button onClick={this.handleAddClick}>Add OEM</Button>
        <FullTable
          ascending={this.state.ascending}
          count={this.state.data.count}
          data={data}
          headers={headers}
          onPageChange={this.handlePageChange}
          onPerPageChange={this.handlePerPageChange}
          onSort={this.handleSort}
          page={this.state.page}
          perPage={this.state.perPage}
          perPageOptions={[5, 10, 25]}
          sortBy={this.state.sortBy}
        />
      </div>
    )
  }
}

export default withRouter(OemsPage)
