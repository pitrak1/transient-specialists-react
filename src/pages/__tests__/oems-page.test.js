import React from 'react'
import { OemsPage } from '../oems-page'
import { shallow } from 'enzyme'
import sinon from 'sinon'
import api from '../../api.js'

describe('OemsPage', () => {
  const data = [
    { id: 1, name: 'OEM 1' },
    { id: 2, name: 'OEM 2' },
    { id: 3, name: 'OEM 3' },
  ]

  it('renders spinner if not yet loaded', () => {
    const node = shallow(<OemsPage />)
    expect(node).toMatchSnapshot()
  })

  it('renders an error if load fails', () => {
    api.getIndex = sinon.stub().callsArgWith(2, 'some error')
    const node = shallow(<OemsPage />)
    expect(node).toMatchSnapshot()
  })

  it('renders data if load succeeds', () => {
    api.getIndex = sinon.stub().callsArgWith(1, data)
    const node = shallow(<OemsPage />)
    expect(node).toMatchSnapshot()
  })

  it('sorts data descending when name column header is clicked', () => {
    api.getIndex = sinon.stub().callsArgWith(1, data)
    const node = shallow(<OemsPage />)
    node.instance().handleSort(null, { id: 'name' })
    expect(node).toMatchSnapshot()
  })

  it('sorts data ascending when name column header is clicked twice', () => {
    api.getIndex = sinon.stub().callsArgWith(1, data)
    const node = shallow(<OemsPage />)
    node.instance().handleSort(null, { id: 'name' })
    node.instance().handleSort(null, { id: 'name' })
    expect(node).toMatchSnapshot()
  })

  it('filters data based on search value', () => {
    api.getIndex = sinon.stub().callsArgWith(1, data)
    const node = shallow(<OemsPage />)
    node.instance().handleSearchChange(null, 'OEM 1')
    expect(node).toMatchSnapshot()
  })
})
