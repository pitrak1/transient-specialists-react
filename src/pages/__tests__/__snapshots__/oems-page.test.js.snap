// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OemsPage filters data based on search value 1`] = `
<div>
  <Flex
    as="span"
    direction="row"
    elementRef={[Function]}
    inline={false}
    justifyItems="start"
    visualDebug={false}
    wrapItems={false}
  >
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={true}
      shrink={true}
    >
      <Heading
        border="none"
        color="inherit"
        ellipsis={false}
        level="h1"
        margin="medium"
      >
        OEMs
      </Heading>
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      margin="small"
      shrink={false}
    >
      <TextInput
        display="block"
        inputContainerRef={[Function]}
        inputRef={[Function]}
        interaction="enabled"
        isRequired={false}
        messages={Array []}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        renderLabel="Search"
        size="medium"
        textAlign="start"
        type="text"
        value="OEM 1"
      />
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      shrink={false}
    >
      <Button
        as="button"
        buttonRef={[Function]}
        cursor="pointer"
        fluidWidth={false}
        margin="0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="default"
      >
        Add OEM
      </Button>
    </FlexItem>
  </Flex>
  <Table
    caption="OEMs"
    hover={true}
    layout="auto"
  >
    <Head>
      <Row>
        <ColHeader
          id="name"
          onRequestSort={[Function]}
          sortDirection="ascending"
          textAlign="start"
        >
          Name
        </ColHeader>
        <ColHeader
          id="Equipment"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
        <ColHeader
          id="Models"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
        <ColHeader
          id="Delete"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
      </Row>
    </Head>
    <Body>
      <Row
        key="1"
      >
        <Cell
          textAlign="start"
        >
          OEM 1
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 1"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Equipment
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 1"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Models
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={1}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Delete
          </Button>
        </Cell>
      </Row>
    </Body>
  </Table>
</div>
`;

exports[`OemsPage renders an error if load fails 1`] = `
<Alert
  liveRegionPoliteness="assertive"
  margin="x-small 0"
  open={true}
  screenReaderOnly={false}
  timeout={0}
  transition="fade"
  variant="error"
>
  some error
</Alert>
`;

exports[`OemsPage renders data if load succeeds 1`] = `
<div>
  <Flex
    as="span"
    direction="row"
    elementRef={[Function]}
    inline={false}
    justifyItems="start"
    visualDebug={false}
    wrapItems={false}
  >
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={true}
      shrink={true}
    >
      <Heading
        border="none"
        color="inherit"
        ellipsis={false}
        level="h1"
        margin="medium"
      >
        OEMs
      </Heading>
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      margin="small"
      shrink={false}
    >
      <TextInput
        display="block"
        inputContainerRef={[Function]}
        inputRef={[Function]}
        interaction="enabled"
        isRequired={false}
        messages={Array []}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        renderLabel="Search"
        size="medium"
        textAlign="start"
        type="text"
        value=""
      />
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      shrink={false}
    >
      <Button
        as="button"
        buttonRef={[Function]}
        cursor="pointer"
        fluidWidth={false}
        margin="0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="default"
      >
        Add OEM
      </Button>
    </FlexItem>
  </Flex>
  <Table
    caption="OEMs"
    hover={true}
    layout="auto"
  >
    <Head>
      <Row>
        <ColHeader
          id="name"
          onRequestSort={[Function]}
          sortDirection="ascending"
          textAlign="start"
        >
          Name
        </ColHeader>
        <ColHeader
          id="Equipment"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
        <ColHeader
          id="Models"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
        <ColHeader
          id="Delete"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
      </Row>
    </Head>
    <Body>
      <Row
        key="1"
      >
        <Cell
          textAlign="start"
        >
          OEM 1
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 1"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Equipment
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 1"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Models
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={1}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Delete
          </Button>
        </Cell>
      </Row>
      <Row
        key="2"
      >
        <Cell
          textAlign="start"
        >
          OEM 2
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 2"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Equipment
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 2"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Models
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={2}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Delete
          </Button>
        </Cell>
      </Row>
      <Row
        key="3"
      >
        <Cell
          textAlign="start"
        >
          OEM 3
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 3"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Equipment
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 3"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Models
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={3}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Delete
          </Button>
        </Cell>
      </Row>
    </Body>
  </Table>
</div>
`;

exports[`OemsPage renders spinner if not yet loaded 1`] = `
<Spinner
  as="div"
  renderTitle="Loading"
  size="large"
  variant="default"
/>
`;

exports[`OemsPage sorts data ascending when name column header is clicked twice 1`] = `
<div>
  <Flex
    as="span"
    direction="row"
    elementRef={[Function]}
    inline={false}
    justifyItems="start"
    visualDebug={false}
    wrapItems={false}
  >
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={true}
      shrink={true}
    >
      <Heading
        border="none"
        color="inherit"
        ellipsis={false}
        level="h1"
        margin="medium"
      >
        OEMs
      </Heading>
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      margin="small"
      shrink={false}
    >
      <TextInput
        display="block"
        inputContainerRef={[Function]}
        inputRef={[Function]}
        interaction="enabled"
        isRequired={false}
        messages={Array []}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        renderLabel="Search"
        size="medium"
        textAlign="start"
        type="text"
        value=""
      />
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      shrink={false}
    >
      <Button
        as="button"
        buttonRef={[Function]}
        cursor="pointer"
        fluidWidth={false}
        margin="0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="default"
      >
        Add OEM
      </Button>
    </FlexItem>
  </Flex>
  <Table
    caption="OEMs"
    hover={true}
    layout="auto"
  >
    <Head>
      <Row>
        <ColHeader
          id="name"
          onRequestSort={[Function]}
          sortDirection="ascending"
          textAlign="start"
        >
          Name
        </ColHeader>
        <ColHeader
          id="Equipment"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
        <ColHeader
          id="Models"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
        <ColHeader
          id="Delete"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
      </Row>
    </Head>
    <Body>
      <Row
        key="1"
      >
        <Cell
          textAlign="start"
        >
          OEM 1
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 1"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Equipment
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 1"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Models
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={1}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Delete
          </Button>
        </Cell>
      </Row>
      <Row
        key="2"
      >
        <Cell
          textAlign="start"
        >
          OEM 2
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 2"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Equipment
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 2"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Models
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={2}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Delete
          </Button>
        </Cell>
      </Row>
      <Row
        key="3"
      >
        <Cell
          textAlign="start"
        >
          OEM 3
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 3"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Equipment
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 3"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Models
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={3}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Delete
          </Button>
        </Cell>
      </Row>
    </Body>
  </Table>
</div>
`;

exports[`OemsPage sorts data descending when name column header is clicked 1`] = `
<div>
  <Flex
    as="span"
    direction="row"
    elementRef={[Function]}
    inline={false}
    justifyItems="start"
    visualDebug={false}
    wrapItems={false}
  >
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={true}
      shrink={true}
    >
      <Heading
        border="none"
        color="inherit"
        ellipsis={false}
        level="h1"
        margin="medium"
      >
        OEMs
      </Heading>
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      margin="small"
      shrink={false}
    >
      <TextInput
        display="block"
        inputContainerRef={[Function]}
        inputRef={[Function]}
        interaction="enabled"
        isRequired={false}
        messages={Array []}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        renderLabel="Search"
        size="medium"
        textAlign="start"
        type="text"
        value=""
      />
    </FlexItem>
    <FlexItem
      as="span"
      elementRef={[Function]}
      grow={false}
      shrink={false}
    >
      <Button
        as="button"
        buttonRef={[Function]}
        cursor="pointer"
        fluidWidth={false}
        margin="0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="default"
      >
        Add OEM
      </Button>
    </FlexItem>
  </Flex>
  <Table
    caption="OEMs"
    hover={true}
    layout="auto"
  >
    <Head>
      <Row>
        <ColHeader
          id="name"
          onRequestSort={[Function]}
          sortDirection="descending"
          textAlign="start"
        >
          Name
        </ColHeader>
        <ColHeader
          id="Equipment"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
        <ColHeader
          id="Models"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
        <ColHeader
          id="Delete"
          sortDirection="none"
          textAlign="start"
          width="2%"
        />
      </Row>
    </Head>
    <Body>
      <Row
        key="3"
      >
        <Cell
          textAlign="start"
        >
          OEM 3
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 3"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Equipment
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 3"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Models
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={3}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Delete
          </Button>
        </Cell>
      </Row>
      <Row
        key="2"
      >
        <Cell
          textAlign="start"
        >
          OEM 2
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 2"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Equipment
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 2"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Models
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={2}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Delete
          </Button>
        </Cell>
      </Row>
      <Row
        key="1"
      >
        <Cell
          textAlign="start"
        >
          OEM 1
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 1"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Equipment
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-name="OEM 1"
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Show Models
          </Button>
        </Cell>
        <Cell
          textAlign="start"
        >
          <Button
            as="button"
            buttonRef={[Function]}
            cursor="pointer"
            data-id={1}
            fluidWidth={false}
            margin="0"
            onClick={[Function]}
            size="medium"
            type="button"
            variant="default"
          >
            Delete
          </Button>
        </Cell>
      </Row>
    </Body>
  </Table>
</div>
`;
